operator/pkg/util/yaml.go

```diff
@@ -65,13 +67,26 @@ func MarshalWithJSONPB(val proto.Message) (string, error) {
 	return string(yb), nil
 }
 
+var (
+	// stdErrMu is used to temporarily divert stderr to silence a bogus warning from external package.
+	stdErrMu sync.Mutex
+)
+
 // UnmarshalWithJSONPB unmarshals y into out using gogo jsonpb (required for many proto defined structs).
 func UnmarshalWithJSONPB(y string, out proto.Message, allowUnknownField bool) error {
 	jb, err := yaml.YAMLToJSON([]byte(y))
 	if err != nil {
 		return err
 	}
 
+	// Silence stray output to stderr in jsonpb package (issues/22261).
+	stdErrMu.Lock()
+	defer stdErrMu.Unlock()
+	stderr := os.Stderr
+	defer func() { os.Stderr = stderr }()
+	_, w, _ := os.Pipe()
```

Comment:
I would do the os.Pipe() creation with an additional defer on Close()'ing `w` before the defer that re-assign os.Stderr to stderr. Because of the LIFO defer ordering it will clarify that os.Stderr is correctly re-assigned before closing.